src/rgw/rgw_data_sync.cc

```diff
@@ -4387,6 +4435,9 @@ int RGWRunBucketSourcesSyncCR::operate()
         }
       }
     }
+    if (progress) {
+      *progress = *std::min_element(shard_progress.begin(), shard_progress.end());
```

Comment:
for each shard, we'll either get the timestamp we read from its sync status, or the timestamp we last wrote to its sync status. the sync status will either store a) the last timestamp incremental sync wrote, b) the timestamp when full sync started, or c) an empty timestamp if those events happened before upgrade

if we get empty timestamps, we lose the `state->progress_timestamp` filtering optimization in DataSyncSingleEntry, and we'll retry RunBucketSourcesSync each time `state->counter` changes - this is effectively the same behavior we got from `marker_tracker->need_retry()` before

but eventually these timestamps will fill in. and in the general case, we _should_ be seeing changes to these bucket shards, because that's what the datalog entry implies

Reply:
it's also worth noting how this timestamp strategy ties into the planned 'sync fairness' work, where we'll be expiring our mdlog and datalog leases somewhat regularly so other gateways have a chance to take over

when a datalog lease is pending this expiration, we'll stop any associated bucket sync. by tracking the timestamps of their progress, we can potentially retire the datalog/error-repo entries that spawned them. that way the next gateway that gets the lease won't have to retry it